### PR TITLE
Bug/strdup parent

### PR DESCRIPTION
**Pull request by khampton353**

### CODE REVIEW

File: src/rpc_typing.c [hunk 1]
```diff
@@ -950,7 +951,7 @@ rpct_read_event(struct rpct_file *file, struct rpct_interface *iface,
 	if (evt->result == NULL) {
 		rpc_set_last_errorf(EINVAL,
 		    "Cannot instantiate type %s of event %s",
-		    evt->result, name);
+		    (type == NULL) ? "NULL" : type, name);
```
Comment (jceel) [inline src/rpc_typing.c]: if `type` is null then `rpct_read_event()` should bail out earlier.

File: src/rpc_typing.c [hunk 2]
```diff
@@ -1014,15 +1014,15 @@ rpct_read_method(struct rpct_file *file, struct rpct_interface *iface,
 			if (arg_name == NULL) {
 				rpc_set_last_errorf(EINVAL,
 				    "Required 'name' field in argument %d "
-				    "of %s missing", idx, name);
+				    "of %s missing", (int)idx, name);
```
Comment (jceel) [inline src/rpc_typing.c]: `%d` should be replaced with `%zu`

File: src/rpc_typing.c [hunk 3]
```diff
@@ -1014,15 +1014,15 @@ rpct_read_method(struct rpct_file *file, struct rpct_interface *iface,
...
 				return ((bool)false);
 			}
 
 			arg_type = rpc_dictionary_get_string(i, "type");
 			if (arg_type == NULL) {
 				rpc_set_last_errorf(EINVAL,
 				    "Required 'type' field in argument %d "
-				    "of %s missing", idx, name);
+				    "of %s missing", (int)idx, name);
```
Comment (jceel) [inline src/rpc_typing.c]: Same here

File: src/rpc_typing.c (hunk 1 above)
Comment (khampton353) [reply src/rpc_typing.c]: Yeah, I was wondering why the previous line checks for type before calling rpct_instantiate_type - it looked like there was an intent to capture both errors at the same same. Do you want an error message for a null type, or for it to be silently ignored?

Comment (jceel) [reply src/rpc_typing.c]: `type == NULL` means that the user didn't specify a type in the .yaml file and it should be treated like syntax error.

Comment (khampton353) [reply src/rpc_typing.c]: Done

**Pull request merged**